lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -265,10 +287,6 @@ public function create_item_permissions_check( $request ) {
 
 		$post_type = get_post_type_object( $this->post_type );
 
-		if ( ! empty( $request['password'] ) && ! current_user_can( $post_type->cap->publish_posts ) ) {
-			return new WP_Error( 'rest_cannot_publish', __( 'Sorry, you are not allowed to create password protected posts in this post type' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
Hmm, this is _no longer_ needed. It was here before to avoid support for password protected posts, which I now removed.